alt-tab-macos/ui/PreferencesPanel.swift

```diff
@@ -1,110 +1,263 @@
 import Cocoa
+import Foundation
 
-class PreferencesPanel: NSPanel, NSTextViewDelegate {
-    var maxScreenUsage: NSTextView?
-    var windowPadding: NSTextView?
-    var cellPadding: NSTextView?
-    var cellBorderWidth: NSTextView?
-    var maxThumbnailsPerRow: NSTextView?
-    var iconSize: NSTextView?
-    var fontHeight: NSTextView?
-    var interItemPadding: NSTextView?
-    var tabKeyCode: NSTextView?
-    var windowDisplayDelay: NSTextView?
-    var metaKey: NSPopUpButton?
-    var theme: NSPopUpButton?
-    var showOnScreen: NSPopUpButton?
-    var inputsMap = [NSTextView: String]()
+class PreferencesPanel: NSPanel, NSWindowDelegate {
+    let panelWidth = CGFloat(496)
+    let panelHeight = CGFloat(256) // auto expands to content height (but does not auto shrink)
+    let panelPadding = CGFloat(40)
+    var labelWidth: CGFloat { get { // derived convenience variable
+        return (panelWidth - panelPadding) * CGFloat(0.45)
+    }}
+    var windowCloseRequested = false
 
     override init(contentRect: NSRect, styleMask style: StyleMask, backing backingStoreType: BackingStoreType, defer flag: Bool) {
-        super.init(contentRect: contentRect, styleMask: style, backing: backingStoreType, defer: flag)
+        let initialRect = NSRect(x: 0, y: 0, width: panelWidth, height: panelHeight)
+        super.init(contentRect: initialRect, styleMask: style, backing: backingStoreType, defer: flag)
         title = Application.name + " Preferences"
         hidesOnDeactivate = false
-        contentView = makeGridView(makeLabelsAndInputs(), makeWarningLabel())
+        contentView = makeContentView()
+        NSApp.activate(ignoringOtherApps: true)
     }
 
-    private func makeLabelsAndInputs() -> [[NSView]] {
+    override func close() {
+        NSApp.deactivate()
+        (NSApp as! Application).preferencesPanel = nil
+        super.close()
+    }
```

Comment:
The easiest (less code) way to let the panel display the actual values from Preferences instead of whatever the user has entered the last time he had the window open. This was a problem for the use-case, when the Preferences rejected the provided value (e.g. CGFloat exception) the PreferencesPanel was still showing the user input (which was rejected) so the the user would think his choice has been set correctly. 

I had many iterations with and without validation and this close (de-instantiation) and new PreferencesPanel instance creation solved this issue without requiring additional code.